src/data-reporting-quicksight-stack.ts

```diff
@@ -120,6 +129,28 @@ export class DataReportingQuickSightStack extends Stack {
       ),
     });
 
+    const realtimeTemplateId = `clickstream_template_rt_${stackParams.redshiftDBParam.valueAsString}_${getShortIdOfStack(Stack.of(this))}`;
+    const realtimeTemplate = new CfnTemplate(this, 'Clickstream-Template-Def-Realtime', {
+      templateId: realtimeTemplateId,
+      awsAccountId: Aws.ACCOUNT_ID,
+      permissions: [{
+        principal: stackParams.quickSightOwnerPrincipalParam.valueAsString,
+        actions: [
+          'quicksight:UpdateTemplatePermissions',
+          'quicksight:DescribeTemplatePermissions',
+          'quicksight:DescribeTemplate',
+          'quicksight:DeleteTemplate',
+          'quicksight:UpdateTemplate',
+        ],
+      }],
+
+      definition: Fn.conditionIf(realtimeDashboardCondition.logicalId,
```

Comment:
use `      definition: JSON.parse(readFileSync(join(__dirname, 'reporting/private/template-def-realtime.json')).toString('utf-8')),` instead

The definition is strong typing. You need to update the required fields to camel style.